### ISSUE TITLE
Upgrading is messy, confusing, and a bit scary

### ISSUE THREAD

**Issue by TacoByte** [NebulousLabs/Sia-UI#296]
RC2 Darwin

I imported the consensus and wallet files before starting the RC2.

It hung for a few minutes before managing to boot into the main screen. Now it's been stuck on "Loading Sia". I guess I'll just wait a bit longer? Peeking inside the Sia folder, I can tell thing are happening as some files' last modified timestamps match the current time. I'm assuming things are going in the right direction and the program is not just stuck in a loop... Or is it? No way to tell for sure.

There needs to be more feedback for the user. I can't determine whether Sia has crashed or what is happening. In-app upgrading is also needed, messing with wallets and other files is not nice and potentially a bit dangerous if you're careless or easily distracted.

**Comment by TacoByte** [comment]
It's slowly calculating the usual folders and files. transactionpool, miner, gateway, host, renter etc... Not sure why so long since I'm not doing any renting/mining/farming. Right now it looks like it's going through the whole blockchain and refreshing contractor.json as it reads the contracts. Perhaps I should have moved the rest of the folders/files as well?

I'm even wondering if this might have an impact on uptime calculations as the whole process is probably going to clock in at around 1 hour. If this is going to be the case for every upgrade, doesn't that leave significantly less time for other kinds of planned and unplanned downtime? If you upgrade twice a month (not unusual), that's a whole day in the year spent upgrading!

If at all possible, I would suggest this to be a 1.0 requisite (proper feedback + in-app upgrading)

**Comment by avahowell** [comment]
Yeah, `siad` can take a long time to rescan an incomplete data-set.  The UI shows "Loading..." when siad has been started but is not finished loading modules, "Welcome Back" when it finishes, and an error if it fails.  There's a planned upgrade to load the API module first, which means the UI can provide more granular feedback on this process.

As for upgrading, the plan is to use a common, OS-specific, data directory so users can just replace their binary and not have to worry about meddling with Sia data.  That issue is here: https://github.com/NebulousLabs/Sia-UI/issues/243 .  It should be implemented for 1.0.  In-app upgrades are much harder, but also planned https://github.com/NebulousLabs/Sia-UI/issues/255.

Thanks again for your feedback.

**Comment by TacoByte** [comment]
Thank you for listening!

I remember the days when Sia used the OS specific directory. Some people did lose their wallets that way, you have to be very careful, especially about multiple clients launching in parallel. Mist (Ethereum) seems to handle this properly, so I'm sure there's a good/safe way to do it.

Perhaps you could even have Sia check for and pull the update from a signed/verified address on the Sia network, so you don't have to deal with a "centralised" (boo!) process of querying github and whatnot.

**Comment by DavidVorick** [comment]
What is the status on this issue? Is the upgrade complete?

Should we consider rescan progress / loading progress to be a part of this issue, or should we open up a separate issue?

**Comment by avahowell** [comment]
Users upgrading from 1.0.0 do not have to manually move their sia data.  The new version of Sia will load their data from the OS-specific data directory, meaning the upgrade process is as simple as launching the new binary.  Rescanning should not be an issue for users upgrading from 1.0.0+.

Closing this issue.
